core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedResultCollector.java

```diff
@@ -67,17 +69,38 @@
   int noDictionaryColumnIndex;
   int complexTypeColumnIndex;
 
+  int noDictionaryComplexColumnIndex = 0;
```

Comment:
All these class variables not needed, move to method level

Reply:
Cannot be moved as it is used at class level

Reply:
Why it cannot be moved?